src/URIs.jl

```diff
@@ -380,6 +380,45 @@ function splitpath(p::AbstractString)
     return elems
 end
 
+"""
+    URIs.normpath(url)
+
+Normalize the path portion of a URI by removing dot segments.
+
+Refer to:
+* https://tools.ietf.org/html/rfc3986#section-5.2.4
+"""
+normpath(url::URI) =
+    URI(scheme=url.scheme, userinfo=url.userinfo, host=url.host, port=url.port,
+        path=normpath(url.path), query=url.query, fragment=url.fragment)
+
+function normpath(p::AbstractString)
+    if isempty(p) || p == "/"
+        return p
+    elseif p == "." || p == ".."
+        return "/"
+    end
+    buf = String[]
+    for part in splitpath(p)
+        if part == "."
+            continue
+        elseif part == ".."
+            isempty(buf) || pop!(buf)
+        else
+            push!(buf, part)
+        end
+    end
+    out = join(buf, '/')
+    # Preserve leading and trailing slashes if present, but don't duplicate them
+    if startswith(p, '/') && !startswith(out, '/')
```

Comment:
I am not sure how one can get a leading `/` in the `out`.
Can you add a test case that causes this? (or comment one that is already there)

Reply:
Dunno, it's just a sanity check. I'm not sure how it would happen.

Reply:
perhaps change it to an `@assert` ?